src/GslibIO.jl

```diff
@@ -52,28 +80,51 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
-
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  dataspec = parse_legacy(filename)                     
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # handle missing values
+  replace!(dataspec.data, na=>NaN)
 
-    # read property values
-    X = readdlm(fs)
+  # create data dictionary
+  table = (; zip(dataspec.varnames, eachcol(dataspec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # handle missing values
-    replace!(X, na=>NaN)
+  georef(table, domain)
+end
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    georef(data, domain)
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  dataspec = parse_legacy(filename)
+
+  # handle missing values
+  replace!(dataspec.data, na=>NaN)
+
+  # we need to identify and separate coordinates and actual attributes 
+  # from the parsed variables
+
+  # find the position of each `coordnames` in `dataspec.varnames`
+  coordpos = []
+  for (i, coordname) in enumerate(coordnames)
+    pos = findall(x -> x == coordname, dataspec.varnames)
+    if isempty(pos)
+      @error "The coordinate name '$coordname' could not be found in the file"
+    else
+      append!(coordpos, pos)
+    end
   end
```

Comment:
Please implement this with `indexin`, it should be one line.

Reply:
It seems `indexin` need the second argument not to be a tuple. For example:
```
indexin([1,2,3,4,5],(2,3))
``` 
fails.

What is the best way to convert coordinates to an iterator? `collect`?

Reply:
Yes, I think `collect` would be fine in this case. 👍🏽 It is not a tight loop that we run multiple times so we can simply allocate a vector here.

Reply:
collect does the work.